pkg/storage/unified/resource/server.go

```diff
@@ -646,6 +661,27 @@ func (s *server) Create(ctx context.Context, req *resourcepb.CreateRequest) (*re
 	ctx, span := tracer.Start(ctx, "resource.server.Create")
 	defer span.End()
 
+	// check quotas and log for now
+	if s.quotaService != nil {
```

Comment:
Did you consider designing the QuotaService so that it performs the validation internally, so it doesn't expose details about eg. the quota struct? IMO designing it that way also makes it feel more like a middleware, which fits well conceptually.

Reply:
Btw, in Go it's easy to move `nil` checks into the methods, and make `nil` value work just fine! This is what protobuf generated messages do (e.g. [getters in `ResourceSearchRequest`](https://github.com/grafana/grafana/blob/002f46736a6683d2d8de8116283341b365ea3210/pkg/storage/unified/resourcepb/search.pb.go#L206-L239)), or our recently added [SearchStats struct](https://github.com/grafana/grafana/blob/42db6324c95cc5713db36eebdcb2a068e20e0cef/pkg/storage/unified/resource/stats.go). There's no need to introduce interface to implement null-object pattern in Go. (Note: while you can add `nil` checks into methods, accessing field directly will still panic.)